### DISCUSSION TITLE
SPI device

### DISCUSSION THREAD

**Discussion by infrafast** [pschatzmann/arduino-audio-tools#1630] [Q&A] [answered]
Hello this library looks fantastic. My device is equiped with a TFT screen, which uses SPI interface. I do believe this is quite a common setup. 

my understanding is that most of the audio component that are driven by this library uses I2S bus. (I mean the ADC and DAC that acquire and stream the audio)

Since SPI already consume 4 pins, it doesn't let enough pin on small footprint MCU (like ESP32S3 from Xiao)  to connect I2S device, so my assumption is that it could prevent from designing product with a lot of TFT screen, limiting the capability to OLED or other I2S solution.

So my question is :
- is it possible to use the library with SPI audio devices and wuold you have example of such realization ?
Thanks a lot.

**Comment by pschatzmann** [comment, marked as answer]
If you use a stereo analog or pwm output you only need 2 pins! ...and for mono only 1 pin.

If you want to use SPI you could use a [VS1053 module like documented here](https://github.com/pschatzmann/arduino-audio-tools/wiki/Audio-Boards#vs1053-audio-codec-modules)

For I2S output you only need 3 free pins.

ps. I just noticed: There's no DAC on the ESP32-S3, so analog output will not work there...

**Reply by infrafast** [reply]
Wow that's a very fast reply :-) thank you I will look into that.

I thought about the one pin for analog input mono that's fine, but as you pointed out there is no DAC so , in my case where you need in and out , a dedicated audio chip is definitely needed.  

Thank you again!
